bool is_moving
uint8 state

uint8 CLOSED=0
uint8 OPENING=1
uint8 OPEN=2
uint8 CLOSING=3